extension/storage/filestorage/extension.go

```diff
@@ -69,16 +61,16 @@ func (lfs *localFileStorage) GetClient(ctx context.Context, kind component.Kind,
 		rawName = fmt.Sprintf("%s_%s_%s_%s", kindString(kind), ent.Type(), ent.Name(), name)
 	}
 	// TODO sanitize rawName
-	absoluteName := filepath.Join(lfs.directory, rawName)
-	client, err := newClient(absoluteName, lfs.timeout)
+	absoluteName := filepath.Join(lfs.cfg.Directory, rawName)
+	client, err := newClient(lfs.logger, absoluteName, lfs.cfg.Timeout, lfs.cfg.Compaction)
 
 	// return if compaction is not required
-	if err != nil || !lfs.compactOnStart {
-		return client, err
+	if err == nil && lfs.cfg.Compaction.OnStart {
+		err = client.Compact(ctx, lfs.cfg.Compaction.Directory, lfs.cfg.Timeout, lfs.cfg.Compaction.MaxTransactionSize)
```

Comment:
We should make sure `.Compact()` always leaves the client in a good state, and then ignore this error after logging it. A failed compaction (usually because of insufficient disk space, I bet) shouldn't prevent us from continuing to use the client.